### PR TITLE
AAP-30888: CVE-2024-45590 ansible-lightspeed-container: Denial of Service Vulnerability in body-parser [ansible_automation_platform-2.4]

### PR DESCRIPTION
**Pull request by manstis**
Jira Issue: https://issues.redhat.com/browse/AAP-30888

## Description
This PR bumps `body-parser` to `1.20.3` (the fixed version).

It was achieved by running `npm audit fix` that also resolved other issues.

## Testing
Re-compiled the webapp.

Ran locally and verified operation.

### Steps to test
1. Pull down the PR`
2. `cd ./ansible_ai_connect_admin_portal`
3. `npm install`
4. `cd ..`
5. `make admin-portal-bundle`
6. `make start-backends`
7. `make create-application`
8. Run server
9. Login with a User that has Admin permissions
10. Navigate to "Admin Portal".

### Scenarios tested
As above.

## Production deployment
- [x] This code change is ready for production on its own
- [ ] This code change requires the following considerations before going to production:

### CODE REVIEW

**Comment by romartin** [comment]
@manstis  I am not sure if only updating the `package-lock.json` is a good practice here, as next time someone does an `npm install `will replace again the `package-lock.json`, right?

What about using `--save` when running `npm audit` fix in order to try to preserve this dependency and its version?

T

**Comment by romartin** [comment]
sorry closed by mistake

**Comment by manstis** [comment]
@romartin 

> `npm install `will replace again the `package-lock.json`, right?

AFAIK the `package-lock.json` only gets re-created if it does not exist.

This is why we check it into GitHub.

File: .snyk [hunk 1]
```diff
@@ -72,4 +72,10 @@ ignore:
           https://www.cve.org/CVERecord?id=CVE-2023-49297
           https://github.com/iterative/PyDrive2/commits/1.20.0
         created: 2024-08-19T08:29:20Z
+  SNYK-PYTHON-ANSIBLECORE-7981515:
+    - '*':
+        reason: |
+          There is no fix yet.
```
Comment (goneri) [inline .snyk:78]: I would also say that the CVE doesn't apply to our use-case.

**Review by goneri — APPROVED** [review]

**Review by goneri — APPROVED** [review]

**Review by romartin — APPROVED** [review]

**Pull request merged**